inferno-ml-server/src/Inferno/ML/Server/Inference.hs

```diff
@@ -235,7 +252,67 @@ runInferenceParam ipid (fromMaybe 128 -> res) =
     withTimeoutMillis :: (Int -> RemoteM b) -> RemoteM b
     withTimeoutMillis =
       (view (#config . #timeout) >>=)
-        . (. (* 1000000) . fromIntegral)
+        . (. (* 1_000_000) . fromIntegral)
+
+    withEvaluationInfo :: RemoteM a -> RemoteM a
+    withEvaluationInfo f = withRunInIO $ \r -> do
+      -- So allocation counter doesn't go below the lower limit, which is
+      -- unlikely but should be accounted for at any rate
+      setAllocationCounter maxBound
+      start <- getCurrentTime
+      bytes0 <- getAllocationCounter
```

Comment:
An alternative to this would be to get one of the memory-related fields from `GHC.Stats.RTSStats` and compare it. But I noticed this was very unreliable unless a GC was triggered before and after -- sometimes the difference between beginning/ending max memory in use was negative. Of course we could trigger a GC before and after eval, but that can cause performance to degrade. So I think getting allocated bytes is sufficient